REFDOC.md

```diff
@@ -0,0 +1,156 @@
+## Class GlobalVars.MSStatus  
+Tracking metasmoke status.  
+### Public interface  
+- `is_up()`: Query if metasmoke is up. Returns `True` if metasmoke is up and `False` otherwise.
+- `is_down()`: Query if metasmoke is down. Returns `True` if metasmoke is down and `False` otherwise.
+- `failed()`: Indicate a metasmoke connection failure.
+- `succeeded()`: Indicate a metasmoke connection success.
+- `get_failure_count()`: Get metasmoke connection failure counter value. The counter counts consecutive failures.
+- `reset_ms_status()`: Reset class `GlobalVars.ms_status` to default values.
+### Thread safety  
+Yes.  
+### Notes on usage  
+- Metasmoke counter represents consecutive metasmoke failures.
+### Implementation  
+This class is implemented with 2 static variables and 1 lock.  
+#### Details  
+##### Attributes  
+- `ms_is_up`: Whether or not metasmoke is up.
+- `counter`: Metasmoke connection failure counter.
+- `rw_lock`: Lock. Controlling access to `ms_is_up`, `failure_count` and `last_ping_time`.
+##### Methods  
+- `set_up()`: Private to `metasmoke.py`. Obtain `rw_lock`. Set `ms_is_up` to `True`. Release `rw_lock`.
+- `set_down()`: Private to `metasmoke.py`. Obtain `rw_lock`. Set `ms_is_up` to `False`. Release `rw_lock`.
+- `is_up()`: Obtain `rw_lock`. Read `ms_is_up` into `current_ms_status`. Release `rw_lock`. Return `current_ms_status`.
+- `is_down()`: Call `is_up()` and return the inverted returned value.
+- `failed()`: Obtain `rw_lock`. Increase `counter` by `1`. Release `rw_lock`.
+- `succeeded()`: Obtain `rw_lock`. Set `counter` to `0`. Release `rw_lock`.
+- `get_failure_count()`: Obtain `rw_lock`. Read `counter` into `current_counter`. Release `rw_lock`. Return `current_failure_count`.
+- `reset_ms_status`: Obtain `rw_lock`. Set `ms_is_up` to `True`. Set `counter` to `0`. Release `rw_lock`.
+#### Considerations  
+- `is_down()` is implemented to call `is_up()`, so maintainance tasks only need to be performed on `is_up()`.
+### Known bugs  
+- None.
+> Last change in this section was on 20 May 2020.
+# File metasmoke.py  
+## Class Metasmoke  
+> Note: the documentation of this class is not yet completed.
+### Subclasses  
+- `AutoSwitch`
+### Public interface  
+- `ms_up()`: Switch metasmoke status to up.
+- `ms_down()`: Switch metasmoke status to down.
+> Not completed yet.
+### Thread safety  
+Unknown. The component currently documented in public interface is thread safe.  
+### Notes on usage  
+- Unless there are good reasons, always call `Metasmoke.ms_up()` and `Metasmoke.ms_down()`
+instead of `GlobalVars.MSStatus.set_up()` and `GlobalVars.MSStatus.set_down()`.
+The former is a wrapper of the latter which offers logging and chat messages.
+### Implementation  
+> Not completed yet.
+### Known bugs  
+- None.
+> Last change in this section was on 20 May 2020.
+## Class Metasmoke.AutoSwitch  
+Automatically switch metasmoke status.
+### Public interface  
+- `to_off()`: Indicate a status ping failure.
```

Comment:
Let's use names for `to_on` and `to_off` that more clearly indicate what they are doing. Just `to_on` and `to_off` are unclear and confusing. Maybe something like `record_ping_success_and_turn_on_ms_if_meet_criteria()` and `record_ping_failure_and_turn_off_ms_if_meet_criteria()`.